packages/vscode-container-client/src/contracts/ContainerClient.ts

```diff
@@ -516,72 +472,38 @@ type TagImageCommand = {
 
 // Inspect Image Command Types
 
-export type InspectImagesItem = {
-    /**
-     * The image ID
-     */
-    id: string;
-    /**
-     * Image name information
-     */
-    image: ImageNameInfo;
-    /**
-     * Repo digest values
-     */
-    repoDigests: string[];
-    /**
-     * Is the image local only?
-     */
-    isLocalImage: boolean;
-    /**
-     * Any environment variables associated with the image
-     */
-    environmentVariables: Record<string, string>;
-    /**
-     * Any default ports exposed by the image
-     */
-    ports: Array<PortBinding>;
-    /**
-     * Any volumes defined by the image
-     */
-    volumes: Array<string>;
-    /**
-     * Any labels assigned to the image
-     */
-    labels: Record<string, string>;
-    /**
-     * The entrypoint for running the image in a container
-     */
-    entrypoint: Array<string>;
-    /**
-     * The command used to start the image in a container
-     */
-    command: Array<string>;
-    /**
-     * The default working directory in the image
-     */
-    currentDirectory?: string;
-    /**
-     * The image architecture
-     */
-    architecture?: 'amd64' | 'arm64';
-    /**
-     * The image operating system
-     */
-    operatingSystem?: ContainerOS;
-    /**
-     * The date the image was created
-     */
-    createdAt?: Date;
-    /**
-     * The default user in the container
-     */
-    user?: string;
-    /**
-     * The RAW inspect output
-     */
-    raw: string;
-};
+export const PortProtocolSchema = z.enum(['udp', 'tcp']);
+export type PortProtocol = z.infer<typeof PortProtocolSchema>;
+
+export const PortBindingSchema = z.object({
+    containerPort: z.number().describe('The internal container port'),
+    hostPort: z.number().optional().describe('The optional host port to bind to the container port'),
+    hostIp: z.string().optional().describe('The optional host IP to bind the port on'),
+    protocol: PortProtocolSchema.optional().describe('The protocol the port uses'),
+});
+
+export type PortBinding = z.infer<typeof PortBindingSchema>;
+
+export const InspectImagesItemSchema = z.object({
+    id: z.string().describe('The image ID'),
+    image: ImageNameInfoSchema.describe('Image name information'),
+    repoDigests: z.array(z.string()).describe('Repo digest values'),
+    isLocalImage: z.boolean().describe('Is the image local only?'),
+    environmentVariables: z.record(z.string(), z.string()).describe('Any environment variables associated with the image'),
+    ports: z.array(PortBindingSchema).describe('Any default ports exposed by the image'),
+    volumes: z.array(z.string()).describe('Any volumes defined by the image'),
+    labels: z.record(z.string(), z.string()).describe('Any labels assigned to the image'),
```

Comment:
We should make a common one for labels and env vars. e.g. `const LabelsSchema = z.record(z.string(), z.string());` and same for env vars, then reference it in these bodies.